x-pack/plugins/apm/public/components/app/settings/agent_configurations/agent_configuration_create_edit/service_page/form_row_select.tsx

```diff
@@ -30,23 +30,44 @@ export function FormRowSelect({
   fieldLabel,
   isLoading,
   options,
-  value,
-  disabled,
+  isDisabled,
   onChange,
 }: Props) {
+  const [selectedOptions, setSelected] = useState<
+    Array<EuiComboBoxOptionOption<string>> | undefined
+  >([]);
+
+  const handleOnChange = (
+    nextSelectedOptions: Array<EuiComboBoxOptionOption<string>>
+  ) => {
+    const [selectedOption] = nextSelectedOptions;
+    setSelected(nextSelectedOptions);
+    onChange(selectedOption.value);
+  };
+
+  useEffect(() => {
+    setSelected(undefined);
+  }, [isLoading]);
+
   return (
     <EuiDescribedFormGroup
       fullWidth
       title={<h3>{title}</h3>}
       description={description}
     >
       <EuiFormRow label={fieldLabel}>
-        <SelectWithPlaceholder
+        <EuiComboBox
+          isClearable={false}
           isLoading={isLoading}
+          singleSelection={{ asPlainText: true }}
           options={options}
-          value={value}
-          disabled={disabled}
-          onChange={onChange}
+          isDisabled={isDisabled}
+          selectedOptions={selectedOptions}
+          onChange={handleOnChange}
+          placeholder={i18n.translate(
```

Comment:
We might want to enable https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md. I was looking for the event handlers (properties starting with `on*`) and having them alphabetised would have made it easier.

We definitely don't want to manually sort these but having an automated eslint fix sounds ok to me.

Reply:
Btw. we can add the eslint rule in a follow PR if others find it useful.